libmscore/score.cpp

```diff
@@ -2887,24 +2887,36 @@ void Score::padToggle(Pad n, const EditData& ed)
       std::vector<ChordRest*> crs;
 
       if (selection().isSingle()) {
-            // do not allow to add a dot on a full measure rest
-            Element* e = selection().element();
-            if (e && e->isRest()) {
-                  Rest* r = toRest(e);
-                  if (r->isFullMeasureRest())
-                        _is.setDots(0);
-                  }
 
-            // on measure rest, select the first actual rest
-            ChordRest* cr = selection().cr();
-            if (cr && cr->isRest() && cr->measure()->isMMRest()) {
-                  Measure* m = cr->measure()->mmRestFirst();
-                  if (m)
-                        cr = m->findChordRest(m->tick(), 0);
+            Element* selectedElement = selection().element();
+
+            if (!(selectedElement->isNote() || selectedElement->isRest())) {
```

Comment:
Maybe we could also build this condition another way. Effectively all this `if` block (or at least [this part](https://github.com/musescore/MuseScore/blob/63445133b5267cbfcb474f3d3bd894f2e48474c7/libmscore/score.cpp#L2898-L2907) of it) defines a condition when clicking note duration button while having a single element selection leads to changing duration of the relevant chord. Currently this condition is indeed equivalent to checking whether the selected element is note or rest, but we can avoid duplicating this condition and move the code starting note input after having checked whether we are able to find the relevant chord, something like
```
if (cr)
      crs.push_back(cr);
else {
      ed.view->startNoteEntryMode();
      // ...
      }
```

Then duplication would be avoided, and it would be much easier to modify this condition if needed.